sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -0,0 +1,221 @@
+        }
+
+        /// <summary>
+        /// Constructs an instance of a <see cref="TableEntity" />.
+        /// </summary>
+        public DynamicTableEntity()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with the specified partition key and row key.
+        /// </summary>
+        /// <param name="partitionKey">A string containing the partition key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        /// <param name="rowKey">A string containing the row key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        public DynamicTableEntity(string partitionKey, string rowKey)
+        {
+            PartitionKey = partitionKey;
+            RowKey = rowKey;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with properties in the <see cref="IDictionary"/>.
+        /// </summary>
+        /// <param name="values">A <see cref="IDictionary"/> containing the initial values of the entity.</param>
+        public DynamicTableEntity(IDictionary<string, object> values)
+        {
+            _properties = values != null ?
+                new Dictionary<string, object>(values) :
+                new Dictionary<string, object>();
+        }
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="String"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public string GetString(string key) => GetValue<string>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="byte"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public byte[] GetBinary(string key) => GetValue<byte[]>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="String"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public bool GetBoolean(string key) => GetValue<bool>(key);
+
+        /// <summary>
+        /// Get the value of a <see cref="DynamicTableEntity"/>'s
+        /// <see cref="DateTime"/> property called
+        /// <paramref name="key"/>.
+        /// </summary>
+        /// <param name="key">The name of the property.</param>
+        /// <returns>The value of the property.</returns>
+        public DateTime GetDateTime(string key) => GetValue<DateTime>(key);
```

Comment:
Does this have to be `DateTime` or could we use `DateTimeOffset` instead?

Reply:
And does something actually transform strings into DateTimes? This one might need a little extra help from us.

Reply:
Currently, the something is [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs#L59)

We could go the custom serialization route if we wanted to streamline things.